app/src/main/java/de/markusfisch/android/shadereditor/widget/SyntaxView.java

```diff
@@ -105,10 +103,8 @@ public void afterTextChanged(Editable s) {
 		textDirty = true;
 		if (ShaderEditorApp.preferences.disableHighlighting() && s.length() > MAX_HIGHLIGHT_LENGTH)
 			return;
-		post(() -> {
-			highlight();
-			invalidate();
-		});
+		highlight();
```

Comment:
Should be synchronized, could lead to crashes, because onDraw is called on another thread